diff '--color=auto' -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2025-10-25 11:20:02.636121539 +0100
+++ b/CMakeLists.txt	2025-10-25 11:27:23.143630586 +0100
@@ -52,6 +52,7 @@
       gfx1100
       gfx1101
       gfx1102
+      gfx1103
       gfx1200
       gfx1201)
 
diff '--color=auto' -Naur a/src/device/common.h b/src/device/common.h
--- a/src/device/common.h	2025-09-29 23:23:07.000000000 +0100
+++ b/src/device/common.h	2025-10-25 11:27:54.039348846 +0100
@@ -26,7 +26,7 @@
   { __atomic_store_n((DST), (SRC), __ATOMIC_SEQ_CST); }
 #endif
 
-#if defined(__gfx1100__) || defined(__gfx1101__) || defined(__gfx1102__) || defined(__gfx1200__) || defined(__gfx1201__)
+#if defined(__gfx1100__) || defined(__gfx1101__) || defined(__gfx1102__) || defined(__gfx1103__) || defined(__gfx1200__) || defined(__gfx1201__)
 #define __trace_hwreg()
 #else
 #define __trace_hwreg() \
diff '--color=auto' -Naur a/src/include/rccl_float8.h b/src/include/rccl_float8.h
--- a/src/include/rccl_float8.h	2025-09-29 23:23:07.000000000 +0100
+++ b/src/include/rccl_float8.h	2025-10-25 11:40:15.027133330 +0100
@@ -45,7 +45,7 @@
 
 #include <hip/hip_fp8.h>
 
-#if   __HIP_DEVICE_COMPILE__ && (defined(__gfx950__) || defined(__gfx1200__) || defined(__gfx1201__) ||  (defined(__gfx1100__) || defined(__gfx1101__)))//HIP_FP8_TYPE_OCP is enabled.
+#if   __HIP_DEVICE_COMPILE__ && (defined(__gfx950__) || defined(__gfx1200__) || defined(__gfx1201__) ||  (defined(__gfx1100__) || defined(__gfx1101__) || defined(__gfx1103__)))//HIP_FP8_TYPE_OCP is enabled.
 typedef __hip_fp8_e4m3 rccl_float8;
 typedef __hip_fp8_e5m2 rccl_bfloat8;
 #elif __HIP_DEVICE_COMPILE__ && (defined(__gfx942__))
diff '--color=auto' -Naur a/tools/JitterBench/Common.hpp b/tools/JitterBench/Common.hpp
--- a/tools/JitterBench/Common.hpp	2025-09-29 23:23:07.000000000 +0100
+++ b/tools/JitterBench/Common.hpp	2025-10-25 11:27:09.987750556 +0100
@@ -43,7 +43,7 @@
 #endif
 
 // Macro for collecting HW_REG_HW_ID
-#if defined(__gfx1100__) || defined(__gfx1101__) || defined(__gfx1102__) || defined(__NVCC__)
+#if defined(__gfx1100__) || defined(__gfx1101__) || defined(__gfx1102__) || defined(__gfx1103__) || defined(__NVCC__)
 #define GetHwId(val) \
   val = 0
 #else
